docs/developer-guide/architecture.md

```diff
@@ -30,15 +43,44 @@ The following controllers for the most part manage a single resource and reflect
 
 The following controllers are more complex and manage multiple resources and reflect those changes in the ngrok API.
 
-### Tunnel Controller
+#### Tunnel Controller
 
 All of the controllers except this tunnel controller use the controller-runtime's Leader Election process so when multiple instances of the controller only 1 is set up to actually try to call the ngrok api to prevent multiple pods from fighting with each other. The tunnel controller is the only controller that does not use this leader election and instead this controller runs in all pods, even non-leaders. This is because this controller is meant to read the Tunnel CRDs created by the ingress controller, and to dynamically manage a list of tunnels using the [ngrok-go](https://github.com/ngrok/ngrok-go) library. It creates these tunnels using labels specified on the Tunnel CRD so they should match an edge's backend created by the ingress controller.
 
 
-### Ingress Controller
+#### Ingress Controller
 
 The ingress controller is the primary piece of functionality in the overall project right now. It is meant to watch Ingress objects and CRDs used by those objects like IPPolicies, NgrokModuleSets, or even secrets.
 
 TODO: Update more about the various pieces of the ingress controller portion such as the store, the driver, how annotations work, etc.
 
 <img src="../assets/images/Under-Construction-Sign.png" alt="Under Construction" width="350" />
+
+## Architecture
+
+### Overview
+
+Previously, in the ngrok Ingress Controller, the Tunnel Controller(non leader-elected) ran under the same manager as the rest of the API controllers(leader-elected). This did not allow users to independetly scale the data forwarding and API management features. Going forward, we will be splitting the Tunnel Controller into its own manager so that it can be scaled independently of the rest of the controllers. This will allow users to scale the data forwarding and API management features independently.
```

Comment:
Nit: I like the historical context, but I feel that the history should come after the present.

Perhaps this paragraph can be 2 paragraphs:
1. The current architectural intent with 2 deployments
2. The historical reasoning

Reply:
I like that format. This will change again once we include the bindings forwarder. I think I'm going to leave it for now since it will get changed soon and reword it for current state once we have the bindings piece